state/controller_test.go

```diff
@@ -142,3 +143,14 @@ func (s *ControllerSuite) TestRemovingUnknownName(c *gc.C) {
 	err := s.State.UpdateControllerConfig(nil, []string{"dr-worm"})
 	c.Assert(err, gc.ErrorMatches, `unknown controller config setting "dr-worm"`)
 }
+
+func (s *ControllerSuite) TestUpdateControllerConfigChecksSpace(c *gc.C) {
+	m, err := s.State.AddMachine("quantal", state.JobManageModel, state.JobHostUnits)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(m.SetMachineAddresses(network.NewAddress("192.168.9.9")), jc.ErrorIsNil)
+
+	err = s.State.UpdateControllerConfig(map[string]interface{}{
+		controller.JujuManagementSpace: "mgmt-space",
+	}, nil)
+	c.Assert(err, gc.ErrorMatches, `invalid config value for "juju-mgmt-space": machine "0" has no addresses in space`)
```

Comment:
I'd probably put the name of the space here.

`invalid config value for "juju-mgmt-space": machine "0" has no addresses in space "mgmt-space"`

That way you see both what machine has an error, and what it has an error with.

Reply:
Done.